cheatsheets/Clickjacking_Defense_Cheat_Sheet.md

```diff
@@ -311,3 +313,166 @@ Activate [designMode](https://developer.mozilla.org/en-US/docs/Web/API/Document/
 ```javascript
 document.designMode = "on";
 ```
+
+## Double Clickjacking
+
+### Introduction
+
+**Double Clickjacking** is an advanced variant of Clickjacking that exploits rapid user interactions by requiring two consecutive clicks to execute a malicious action. Unlike traditional Clickjacking, which typically relies on a single deceptive click, Double Clickjacking increases the attack success rate by bypassing single-click protections and evading common security measures.
+
+## Attack Scenario
+
+1. The attacker loads a malicious webpage containing a transparent iframe overlaying a legitimate website.
+
+2. The user is tricked into clicking an element (e.g., a button or link), thinking it belongs to the attacker’s site.
+
+3. The first click moves the transparent iframe into position over a critical UI element of the target website.
+
+4. The second click executes a malicious action, such as transferring funds, changing security settings, or posting content without user consent.
+
+## Ineffective Defenses
+
+Traditional Clickjacking protections are ineffective against Double Clickjacking, as noted in research by Paulos Yibelo:
+
+### 1. **X-Frame-Options:**
+
+  Attackers manipulate iframe positions dynamically, making this defense ineffective.
+
+### 2. **Content Security Policy (CSP) Frame-Ancestors:**
+
+This protection does not prevent UI manipulation via JavaScript or CSS.
+
+### 3. **SameSite Cookies:**
+
+Attackers do not need cross-site requests, rendering this defense ineffective.
+
+## Advanced Mitigation Strategies
+
+Given the shortcomings of traditional defenses, the following advanced approaches offer stronger protection against Double Clickjacking:
+
+### **1. Detect Hidden Iframes Using Intersection Observer**
+
+**Concept:**  Attackers often use transparent or off-screen iframes. This detects and alerts users if such an iframe is capturing interactions.
+
+```javascript
+const observer = new IntersectionObserver(entries => {
+    entries.forEach(entry => {
+        if (!entry.isIntersecting) {
+            alert("Warning: A hidden iframe may be attempting a Clickjacking attack!");
+        }
+    });
+}, { threshold: 0.5 });
```

Comment:
@jmanico - I am not enough of a JavaScript expert to be able to discern if this is an effective defense against Double Clickjacking or not. Could you please take a look at this and give us your thoughts regarding it's effectiveness? It is very different than Yibelo's client-side JS defense. And I am a little concerned that this may have false positives (the assumption seemingly being it is rare for multiple entities to overlap) and also it may be a time-consuming defense if there are a lot of entities on a page, given that you'd need to check this on every page.